lib/puppet/module.rb

```diff
@@ -423,32 +423,22 @@ def strict_semver?
   end
 
   def initialize_i18n
-    module_name = @forge_name.gsub("/","-") if @forge_name
-    return if module_name.nil? || i18n_initialized?(module_name)
+    module_name = @forge_name ? @forge_name.gsub("/", "-") : name
+    return if Puppet::GettextConfig.translations_loaded?(module_name)
 
     locales_path = File.absolute_path('locales', path)
 
-    if Puppet::GettextConfig.initialize(locales_path, :po)
-      Puppet.debug "#{module_name} initialized for i18n: #{GettextSetup.translation_repositories[module_name]}"
+    if Puppet::GettextConfig.load_translations(module_name, locales_path, :po)
```

Comment:
Wherever we talk about translating modules, we should make it clear the translation file name needs to match the module namespace-name.